modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition.py

```diff
@@ -124,3 +107,28 @@ def put(cls, obj):
             The new partition.
         """
         return cls(obj)
+
+    @property
+    def _length_cache(self):
+        """
+        Number of rows.
+
+        Returns
+        -------
+        int
+        """
+        return len(self._data)
+
+    @property
+    def _width_cache(self):
+        """
+        Number of columns.
+
+        Returns
+        -------
+        int
+        """
+        if isinstance(self._data, pa.Table):
+            return self._data.num_columns
+        else:
+            return self._data.shape[0]
```

Comment:
is shape[0] the correct way to access the number of columns? not shape[1]?

Reply:
You are right, it should be shape[1]. I'll fix it in a separate PR.

Reply:
https://github.com/modin-project/modin/pull/6353